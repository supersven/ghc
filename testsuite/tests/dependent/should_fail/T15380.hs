{-# LANGUAGE TypeFamilies #-}
{-# LANGUAGE TypeInType #-}
{-# LANGUAGE TypeOperators #-}

module T15380 where

import Data.Kind

class Generic a where
  type Rep a :: Type

class PGeneric a where
  type To a (x :: Rep a) :: a

type family MDefault (x :: a) :: a where
  MDefault x = To (M x)

class C a where
  type M (x :: a) :: a
  type M (x :: a) = MDefault x
