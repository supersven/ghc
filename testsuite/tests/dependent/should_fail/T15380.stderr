
T15380.hs:16:16: error:
    • Expecting one more argument to ‘To (M x)’
      Expected a type, but ‘To (M x)’ has kind ‘Rep (M x) -> M x’
    • In the type ‘To (M x)’
      In the type family declaration for ‘MDefault’
